architxt/schema.py

```diff
@@ -141,6 +141,51 @@ def relations(self) -> dict[NodeLabel, tuple[NodeLabel, NodeLabel]]:
             if has_type(production, NodeType.REL)
         }
 
+    def get_relations_type(self, forest: Forest) -> dict[NodeLabel, tuple[NodeLabel, NodeLabel]]:
```

Comment:
This should be computed inside `Schema.from_forest`, the information should then be retrieve by `Schema.relations` changing the signature to return a dataclass instead of a dict to include extra informations.